src/genomic_features/ensembl/ensembldb.py

```diff
@@ -115,15 +116,214 @@ def __init__(self, connection: ibis.BaseBackend):
+        return result
 
     def chromosomes(self):
         """Get chromosome information."""
         return self.db.table("chromosome").execute()
+
+    def build_query(self, table, cols, filter, join_type="inner"):
+        """Build a query for the genomic features table."""
+        # check if join is required
+        tables = self.get_required_tables(self.tables_for_columns(cols))
+        if len(tables) > 1:
+            query = self.join_query(tables, start_with=table, join_type=join_type)
+        else:
+            query = self.db.table(table)
+        # add filter
+        query = query.filter(filter.convert()).select(cols)
+        return query
+
+    def join_query(self, tables, start_with, join_type="inner"):
+        """Join tables and return a query."""
+        # check for intermediate tables
+
+        tables.remove(start_with)
+        tables = [start_with] + tables
+        print(tables)
+        if join_type == "inner":
+            query = self.inner_join_query(tables)
+        elif join_type == "left":
+            query = self.left_join_query(tables)
+        else:
+            raise ValueError(f"Invalid join type: {join_type}")
+
+        return query
+
+    def inner_join_query(self, tables):
+        """Join tables in inner join and return a query."""
+        # build query
+        query = self.db.table(tables[0])
+        for t in tables[1:]:
+            # determine common columns
+            # TODO: Determine if we want to use all common columns
+            common_cols = list(set(query.columns) & set(self.db.table(t).columns))
+            query = query.join(self.db.table(t), predicates=common_cols, how="inner")
+        return query
+
+    def left_join_query(self, tables):
+        """Join tables in left join and return a query."""
+        # build query
+        query = self.db.table(tables[0])
+        for t in tables[1:]:
+            # determine common columns
+            common_cols = list(set(query.columns) & set(self.db.table(t).columns))
+            query = query.join(
+                self.db.table(t),
+                predicates=common_cols,
+                how="left",
+                suffixes=("", "_y"),
+            )
+            query = query.drop(
+                *[f"{c}_y" for c in common_cols]
+            )  # drop duplicate columns
+        return query
+
+    def list_tables(self) -> list:
+        """List all tables available in the genomic features database."""
+        return self.db.list_tables()
+
+    def tables_by_degree(self, tab: list[str] = None) -> list:
+        """Order tables available in the genomic features database."""
+        if tab is None:
+            tab = self.list_tables()  # list of table names
+        # check that all tables are in the database and print warning
+        if not set(tab).issubset(set(self.list_tables())):
+            print(
+                f"Warning: tables {set(tab) - set(self.list_tables())} are not in the database."
+            )
```

Comment:
Should this be an error?

If it should be a warning, how about [`warnings.warn`](https://docs.python.org/3/library/warnings.html#warnings.warn) instead?

Reply:
At the moment it automatically deletes the tables that are not in the database and prints a warning (which I can replace by `warnings.warn`) unless you think error is preferable